File: contrib/sdk-bindings/rust/python-tests/test_types.py
# SPDX-FileCopyrightText: 2025 Functori <contact@functori.com>
#
# SPDX-License-Identifier: MIT

from tezos import (
    BigInt,
)

def test_big_int():
    big_int_from_int = BigInt.from_int(2**63 - 1)
    big_int_from_string = BigInt.from_string("9223372036854775807")

    assert big_int_from_int == big_int_from_string

    other_big_int_from_string = BigInt.from_string("9223372036854775808")

    assert big_int_from_int != other_big_int_from_string

File: contrib/sdk-bindings/kotlin/demo/app/src/test/kotlin/org/example/TestTypes.kt
/* SPDX-FileCopyrightText: 2025 Functori <contact@functori.com>
 *
 * SPDX-License-Identifier: MIT
 */

package org.example.tezos

import kotlin.test.Test
import kotlin.test.assertEquals
import kotlin.test.assertNotEquals

class Types {

    @Test
    fun testBigInt() {
        val bigIntFromInt = BigInt.fromInt(Long.MIN_VALUE)
        val bigIntFromString = BigInt.fromString("-9223372036854775808")

        assertEquals(bigIntFromInt, bigIntFromString)

        val otherBigIntFromString = BigInt.fromString("-9223372036854775809")

        assertNotEquals(bigIntFromInt, otherBigIntFromString)
    }

}
